bridge/github/import_test.go

```diff
@@ -0,0 +1,223 @@
+					bug.NewAddCommentOp(author, 0, "first comment", nil),
+					bug.NewEditCommentOp(author, 0, "", "first comment edited", nil),
+				},
+			},
+		},
+		{
+			name:  "comment deletion",
+			exist: true,
+			url:   "https://github.com/MichaelMure/git-but-test-github-bridge/issues/5",
+			bug: &bug.Snapshot{
+				Operations: []bug.Operation{
+					bug.NewCreateOp(author, 0, "comment deletion", "", nil),
+				},
+			},
+		},
+		{
+			name:  "edition deletion",
+			exist: true,
+			url:   "https://github.com/MichaelMure/git-but-test-github-bridge/issues/6",
+			bug: &bug.Snapshot{
+				Operations: []bug.Operation{
+					bug.NewCreateOp(author, 0, "edition deletion", "initial comment", nil),
+					bug.NewEditCommentOp(author, 0, "", "initial comment edited again", nil),
+					bug.NewAddCommentOp(author, 0, "first comment", nil),
+					bug.NewEditCommentOp(author, 0, "", "first comment edited again", nil),
+				},
+			},
+		},
+		{
+			name:  "hidden comment",
+			exist: true,
+			url:   "https://github.com/MichaelMure/git-but-test-github-bridge/issues/7",
+			bug: &bug.Snapshot{
+				Operations: []bug.Operation{
+					bug.NewCreateOp(author, 0, "hidden comment", "initial comment", nil),
+					bug.NewAddCommentOp(author, 0, "first comment", nil),
+				},
+			},
+		},
+		{
+			name:  "transfered issue",
+			exist: true,
+			url:   "https://github.com/MichaelMure/git-but-test-github-bridge/issues/8",
+			bug: &bug.Snapshot{
+				Operations: []bug.Operation{
+					bug.NewCreateOp(author, 0, "transfered issue", "", nil),
+				},
+			},
+		},
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	repo, err := repository.NewGitRepo(cwd, bug.Witnesser)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	backend, err := cache.NewRepoCache(repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer backend.Close()
+	interrupt.RegisterCleaner(backend.Close)
+
+	importer := &githubImporter{}
+	err = importer.Init(core.Configuration{
+		"user":    "MichaelMure",
+		"project": "git-but-test-github-bridge",
+		"token":   os.Getenv("GITHUB_TOKEN"),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = importer.ImportAll(backend, time.Time{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ids := backend.AllBugsIds()
+	assert.Equal(t, len(ids), 8)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b, err := backend.ResolveBugCreateMetadata(keyGithubUrl, tt.url)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ops := b.Snapshot().Operations
+			if tt.exist {
+				assert.Equal(t, len(tt.bug.Operations), len(b.Snapshot().Operations))
+
+				for i, op := range tt.bug.Operations {
+					switch op.(type) {
+					case *bug.CreateOperation:
+						if op2, ok := ops[i].(*bug.CreateOperation); ok {
+							assert.Equal(t, op2.Title, op.(*bug.CreateOperation).Title)
+							assert.Equal(t, op2.Message, op.(*bug.CreateOperation).Message)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = CreationOperation", i)
```

Comment:
you can probably do that once in the loop, and then remove all the type checks and `continue`